CHANGELOG.md

```diff
@@ -1,5 +1,98 @@
+
+    [bradyjoslin]: https://github.com/bradyjoslin
+    [pull/1110]: https://github.com/cloudflare/wrangler/pull/1110
+    [issue/975]: https://github.com/cloudflare/wrangler/issues/975
+
+  - **`wrangler dev` checks if the specified port is available - [EverlastingBugstopper], [issue/1122] [pull/1272]**
+
+    When starting up `wrangler dev`, it now checks to see if the requested port is already in use and returns a helpful error message if that's the case.
+
+    [EverlastingBugstopper]: https://github.com/EverlastingBugstopper
+    [pull/1272]: https://github.com/cloudflare/wrangler/pull/1272
+    [issue/1122]: https://github.com/cloudflare/wrangler/issues/1122
+
+- ### Fixes
+
+  - **Don't reinstall vendored binaries on every build - [EverlastingBugstopper], [issue/768] [pull/1003]**
+
+    You may have noticed some very verbose and over-eager installation output when running Wrangler. Every `webpack` type build would install `wranglerjs` and `wasm-pack`. This was... super annoying and not a great experience, especially when running `wrangler preview --watch` or `wrangler dev`. Each time you'd change a file, Wrangler would reinstall those external dependencies. This doesn't happen anymore! Wrangler will still download and install these external dependencies, but only if you have an outdated version.
+
+    [EverlastingBugstopper]: https://github.com/EverlastingBugstopper
+    [pull/1003]: https://github.com/cloudflare/wrangler/pull/1003
+    [issue/768]: https://github.com/cloudflare/wrangler/issues/768
+
+  - **Remove redundant builds - [EverlastingBugstopper], [issue/1219] [pull/1269]**
+
+    When running `wrangler preview --watch` or `wrangler dev` on a `webpack` type project, Wrangler will provide a new build artifact and upload it via the Cloudflare API. Before, we'd start a long-running `webpack --watch` command, _in addition to_ running `webpack` on every change. We were running two builds on every change! This was not great and has been removed. This, combined with the above fix removing redundant installations, should greatly improve your dev iteration cycles.
+
+    [EverlastingBugstopper]: https://github.com/EverlastingBugstopper
+    [pull/1269]: https://github.com/cloudflare/wrangler/pull/1269
+    [issue/1219]: https://github.com/cloudflare/wrangler/issues/1219
+
+  - **`wrangler dev` will reconnect to the devtools WebSocket after being disconnected - [EverlastingBugstopper], [issue/1241] [pull/1276]**
+
+    `wrangler dev` initiates a WebSocket connection via the Cloudflare API in order to stream `console.log` messages to your terminal. After waiting a while, the Workers runtime will disconnect the WebSocket connection, and Wrangler would panic, requiring developers to restart the process. Now, if `wrangler dev` gets disconnected, it will issue a reconnect request, allowing developers to run `wrangler dev` as long as they are connected to the Internet.
```

Comment:
```suggestion
    `wrangler dev` initiates a WebSocket connection via the Cloudflare API in order to stream `console.log` messages to your terminal. Over time, the WebSocket connection could get disconnected, and Wrangler would panic, requiring developers to restart the process. Now, if `wrangler dev` gets disconnected, it will issue a reconnect request, allowing developers to run `wrangler dev` as long as they are connected to the Internet.
```
Suggest clarifying that this isn't strictly a thing the runtime _will_ do, but that for many reasons, long-lived Websocket connections can be disconnected.